server/routes/classes.ts

```diff
@@ -26,29 +31,24 @@ function getFileNameFromUrl(url: string): string | null {
     return match ? match[1] : null
 }
 
-router.get('/:id', async (req: Request, res: Response) => {
-    //TODO: Check if user has access to class
-
-    const classData = await Class.findById(req.params.id)
-    if (classData) {
-        return res.status(200).json(classData)
+router.get('/:id',authUser, async (req: Request, res: Response) => {
+    try{
+        //TODO: Check if user is enrolled in the course
+        const classData = await Class.findById(req.params.id)
+        if (classData) {
+            return res.status(200).json(classData)
+        }else{
+            return res.status(404).json({ error: ERROR_CLASS_NOT_FOUND })
+        }
+    }catch{
+        return res.status(500).json({ error: ERROR_SERVER })
     }
-    return res.status(404).json({ error: 'Class not found' })
-})
-
-//Reminder
-
-/*
-
-this route lets user just create a new class, but it doesn't let him add it to a course
 
-this is responsible of another microservice
-
-*/
+})
 
 //TODO: Connect to course microservice in frontend
 
-router.post('/', upload.single('file'), async (req: Request, res: Response) => {
+router.post('/',authUser, upload.single('file'), async (req: Request, res: Response) => {
```

Comment:
With respect to the OAS defined, the endpoint associated to the POST must be "/classes/course/{courseId}" to know in which course the class is being uploaded.